src/IQFeed.CSharpApiClient.Tests/Lookup/LookupRateLimiterTests.cs

```diff
@@ -17,26 +17,35 @@ public class LookupRateLimiterTests
         [Test]
         public async Task Should_Rate_Limit_By_Throughput()
         {
+
             // Arrange
-            var totalSeconds = TimeSpan.FromSeconds(15).TotalSeconds;
+            var totalSeconds = TimeSpan.FromSeconds(10).TotalSeconds;
             var seconds = 0;
 
             var requestsPerSecond = 50;
             var requestsCount = 0;
-            var requests = new List<int>();
+            var testResults = new List<TestResult>();
 
             var lookupRateLimiter = new LookupRateLimiter(requestsPerSecond);
             var cts = new CancellationTokenSource();
 
-            // Act
-            _ = Task.Run(async () =>
+            var numberOfClients = 15;
+            var tasks = new Task[numberOfClients];
+            for (var i = 0; i < numberOfClients; i++)
             {
-                while (true)
+                tasks[i] = new Task(async () =>
```

Comment:
@Nucs this is working fine with 15 clients. you see, the output doesnt change if you add 1000 tasks.